crates/sui-test-transaction-builder/src/lib.rs

```diff
@@ -106,16 +130,19 @@ impl TestTransactionBuilder {
         self
     }
 
-    pub fn publish(mut self, path: PathBuf) -> Self {
+    pub fn publish(mut self, path: PathBuf, with_unpublished_deps: bool) -> Self {
```

Comment:
Is with_unpublished_deps ever passed in with a different value?

Reply:
It is. In `sui-source-validation` tests via `context.make_publish_transaction`

Reply:
Consider making this a separate with function, so that publish by default sets it to false. I think false is the most common case.